agent/agent.go

```diff
@@ -275,25 +248,29 @@ func (r *agent) GetConfig() Config {
 
 // Start starts the agent's background tasks.
 func (r *agent) Start() error {
-	errChan := make(chan error, 1)
-
-	go r.statusUpdateLoop()
-	go r.timelineUpdateLoop(context.TODO())
+	// TODO: Modify Start to accept a context.
+	ctx := context.TODO()
+	go r.recycleLoop(ctx)
+	go r.statusUpdateLoop(ctx)
+	go r.timelineUpdateLoop(ctx)
+	go r.serveMetrics(ctx)
+	return nil
+}
 
-	if r.metricsListener != nil {
-		go func() {
-			http.Handle("/metrics", promhttp.Handler())
-			if err := http.Serve(r.metricsListener, nil); err != nil {
-				errChan <- trace.Wrap(err)
-			}
-		}()
+// serveMetrics registers the prometheus metrics handler and starts accepting
+// connections on the metrics listener.
+func (r *agent) serveMetrics(ctx context.Context) {
+	if r.metricsListener == nil {
+		return
 	}
 
-	select {
```

Comment:
Wasn't exactly sure what the purpose of this code was.
When do we care if `http.Serve` returns an error within the first second?

Reply:
I think the intention was to handle the case of http failing to `Serve` immediately which should have failed the whole method - now, this case is simply ignored.